packages/contracts-bedrock/src/L1/OptimismPortal.sol

```diff
@@ -150,6 +161,16 @@ contract OptimismPortal is Initializable, ResourceMetering, ISemver {
         return _byteCount * 16 + 21000;
     }
 
+    /// @notice Retuns the balance of the contract.
+    function balance() public view returns (uint256) {
+        (address token,) = gasPayingToken();
+        if (token == Constants.ETHER) {
+            return address(this).balance;
+        } else {
+            return _balance;
+        }
+    }
```

Comment:
What do you think of having an invariant that when `token == Constants.ETHER`, `address(this).balance >= _balance`. Then we can just always rely on the internal accounting of `_balance` to simplify this method. I'm still reviewing so not convinced yet this is better, just an idea

Reply:
Right now we don't update `_balance` on ether deposits but you are right that we could simplify this. Plus using `BALANCE` opcode isn't semantically correct because you can use selfdestruct or `donateETH()` or possibly `PAY` in the future to send ether to the contract without it being minted on L2

Reply:
I agree with your invariant. An issue with adopting `_balance` on existing chains is that we would need to initialize with a non 0 value

Reply:
should we then update `_balance` on ether deposits? what are the properties it should have when choosing to what value to initiate `_balance` to? @tynes

Reply:
I am in favor of leaving as is, I don't feel like coordinating a way to upgrade such that we are able to get the balances just right in the initialize call